containerd-shim-v2/metrics.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2018 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package containerdshim
+
+import (
+	"github.com/containerd/cgroups"
+	"github.com/containerd/typeurl"
+
+	google_protobuf "github.com/gogo/protobuf/types"
+)
+
+func marshalMetrics(s *service, containerID string) (*google_protobuf.Any, error) {
+	stats, err := s.sandbox.StatsContainer(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	cgStats := stats.CgroupStats
+
+	var hugetlb []*cgroups.HugetlbStat
+	for _, v := range cgStats.HugetlbStats {
+		hugetlb = append(
+			hugetlb,
+			&cgroups.HugetlbStat{
+				Usage:   v.Usage,
+				Max:     v.MaxUsage,
+				Failcnt: v.Failcnt,
+			})
+	}
+
+	var perCPU []uint64
+	for _, v := range cgStats.CPUStats.CPUUsage.PercpuUsage {
+		perCPU = append(perCPU, v)
+	}
+
+	metrics := &cgroups.Metrics{
```

Comment:
Please factorize this in a converting function. Something like `statsToMetrics()` that would take care of converting the stats returned by Kata into the metrics structure expected by the containerd API.